misc/pytorch_toolkit/chest_xray_screening/tools/test.py

```diff
@@ -0,0 +1,135 @@
+import os
+import os.path
+import numpy as np
+import time
+import sys
+import csv
+import argparse
+import torch
+from torchvision import models
+import torch.nn as nn
+import torch.backends.cudnn as cudnn
+import torchvision
+import torchvision.transforms as transforms
+import torch.optim as optim
+import torch.nn.functional as tfunc
+from torch.utils.data import Dataset
+from torch.utils.data.dataset import random_split
+from torch.utils.data import DataLoader
+from torch.optim.lr_scheduler import ReduceLROnPlateau,StepLR
+from PIL import Image
+import torch.nn.functional as func
+import random
+from sklearn.metrics.ranking import roc_auc_score
+import sklearn.metrics as metrics
+import json
+from pathlib import Path
+from dataloader import RSNADataSet
+
+
+class RSNAInference():
+
+    #---- Computes area under ROC curve 
+    #---- dataGT - ground truth data
+    #---- dataPRED - predicted data
+    #---- classCount - number of classes
+    
+    def computeAUROC (dataGT, dataPRED, classCount):
+        
+        outAUROC = []
+        
+        datanpGT = dataGT.cpu().numpy()
+        datanpPRED = dataPRED.cpu().numpy()
+        #print(datanpGT.shape)
+        #print(datanpPRED.shape)
+        
+        for i in range(classCount):
+            try:
+                outAUROC.append(roc_auc_score(datanpGT[:, i], datanpPRED[:, i]))
+            except ValueError:
+                outAUROC.append(0)#when the number of counts is zero
+        return outAUROC
+        
+    
+    def test(model, dataLoaderTest, nnClassCount, checkpoint, class_names,device):   
+        cudnn.benchmark = True
+        
+        if checkpoint != None:
+            modelCheckpoint = torch.load(checkpoint)
+            model.load_state_dict(modelCheckpoint['state_dict'])
+        else:
+            model.state_dict()
```

Comment:
Same as above (in `train.py` file). Don't compare with singletons (like `None`) via comparison operators. Extract loading model into separate function and import it everywhere it is used.